packages/ui-causal-graph-editor/src/shared/use-causal-graph-editor.tsx

```diff
@@ -63,36 +64,36 @@ export default function useCausalGraphEditor(
         }
     );
 
+    // bind each action creator to dispatch
+    const api = useMemo(() => {
+        return actionNames.reduce<GraphApi>((acc, actionName) => {
+            const actionCreator = GraphActionCreators[actionName];
+            // eslint-disable-next-line prefer-spread
+            acc[actionName] = (...args: Parameters<typeof actionCreator>) => dispatch(actionCreator.apply(null, args));
+            return acc;
+        }, {} as GraphApi);
+    }, [dispatch]);
+
     const isTimeSeriesCausalGraph = useMemo(() => {
-        const nodeClass = state.graph.getNodeAttribute(state.graph.nodes()[0], 'extras')?.node_class;
+        const nodeClass = Object.values(graphData.nodes)[0]?.node_class;
         return nodeClass === 'TimeSeriesNode';
-    }, [state.graph]);
+    }, [graphData]);
 
     const layout = useMemo(() => {
-        // If we have a time series causasl graph and if the layout chosen supports tiers and these are not defined
         const newLayout = graphLayout;
+        // If the graph is a time series graph, tiers are not defined and is not a PlanarLayout, we update the tiers to show the time series in layers
         if (
             isTimeSeriesCausalGraph &&
             'tiers' in newLayout &&
             'orientation' in newLayout &&
-            newLayout.tiers === undefined
+            newLayout.tiers === undefined &&
+            !(newLayout instanceof PlanarLayout)
         ) {
-            // We set tiers based on TimeSeriesCausalGraph structure of node lags
-            newLayout.tiers = { group: 'variable_name', order_nodes_by: 'time_lag' };
+            newLayout.tiers = { group: 'extras.time_series_variable', order_nodes_by: 'time_lag' };
```

Comment:
I think we missed it before but `newLayout` is a reference to the same layout object as `graphLayout` which ends up mutating the prop. I think best would be to deep clone the layout when creating the copy.

Since layout is a class instance it will be slightly tricky to clone, but a lodash clone (or `structuredClone`) and maybe also explicitly setting the prototype of the clone should we enough.

Reply:
structuredClone and lodash's didn't quite work because they would loose the notion of the instance of the layout class. So instead I added a clone method to GraphLayout

Reply:
Yeah they don't work by themselves, that's why I suggested you might also need to set the prototype. I wonder if this works, e.g.
```ts
Object.create(
    Object.getPrototypeOf(obj), 
    Object.getOwnPropertyDescriptors(obj) 
)
```
or something in that vein. 
Regardless, your approach is more 'proper' though a bit more verbose and we'll just need to remember to keep the clone method up to date as we add more properties

Reply:
That does work, have updated to use it instead